packages/dev/core/src/Materials/Node/Blocks/Dual/sceneDepthBlock.ts

```diff
@@ -184,6 +186,7 @@ export class SceneDepthBlock extends NodeMaterialBlock {
     protected _buildBlock(state: NodeMaterialBuildState) {
         super._buildBlock(state);
 
+        this._samplerName = "sceneDepthSampler" + ID++;

```

Comment:
That would be better to use something like:

```
this._samplerName = state._getFreeVariableName(this.name + "Sampler");
```
It is what `TextureBlock` is doing.

Reply:
Thanks for the feedback, updated it! :D